src/iam/iam-base.ts

```diff
@@ -245,38 +255,97 @@ export class IAMBase {
       return;
     }
-    this._transactionOverrides = {
-      gasLimit: hexlify(4900000),
-      gasPrice: hexlify(0.1)
-    };
+    if (
+      walletProvider === WalletProvider.EwKeyManager ||
+      walletProvider === WalletProvider.WalletConnect
+    ) {
+      this._transactionOverrides = {
+        gasLimit: hexlify(4900000),
+        gasPrice: hexlify(0.1)
+      };
 
-    const showQRCode = !(walletProvider === WalletProvider.EwKeyManager);
-
-    this._walletConnectProvider = new WalletConnectProvider({
-      rpc: {
-        [this._connectionOptions.chainId]: this._connectionOptions.rpcUrl
-      },
-      infuraId: this._connectionOptions.infuraId,
-      bridge: this._connectionOptions.bridgeUrl,
-      qrcode: showQRCode
-    });
-
-    if (walletProvider === WalletProvider.EwKeyManager) {
-      this._walletConnectProvider.wc.on("display_uri", (err, payload) => {
-        // uri is expected to be WalletConnect Standard URI https://eips.ethereum.org/EIPS/eip-1328
-        const wcUri = payload.params[0];
-
-        const encoded = encodeURIComponent(wcUri);
-        const hasQueryString = this._ewKeyManagerUrl.includes("?");
-        const url = `${this._ewKeyManagerUrl}${hasQueryString ? "&" : "?"}uri=${encoded}`;
-        window.open(url, "_blank");
+      const showQRCode = !(walletProvider === WalletProvider.EwKeyManager);
+
+      this._walletConnectProvider = new WalletConnectProvider({
+        rpc: {
+          [this._connectionOptions.chainId]: this._connectionOptions.rpcUrl
+        },
+        infuraId: this._connectionOptions.infuraId,
+        bridge: this._connectionOptions.bridgeUrl,
+        qrcode: showQRCode
       });
+
+      if (walletProvider === WalletProvider.EwKeyManager) {
+        this._walletConnectProvider.wc.on("display_uri", (err, payload) => {
+          // uri is expected to be WalletConnect Standard URI https://eips.ethereum.org/EIPS/eip-1328
+          const wcUri = payload.params[0];
+
+          const encoded = encodeURIComponent(wcUri);
+          const hasQueryString = this._ewKeyManagerUrl.includes("?");
+          const url = `${this._ewKeyManagerUrl}${hasQueryString ? "&" : "?"}uri=${encoded}`;
+          window.open(url, "_blank");
+        });
+      }
+
+      await this._walletConnectProvider.enable();
+
+      this._walletConnectProvider.wc.on("session_update", () => {
+        this.closeConnection();
+      });
+
+      this._walletConnectProvider.wc.on("disconnect", () => {
+        this.closeConnection();
+      });
+
+      this._signer = new providers.Web3Provider(this._walletConnectProvider).getSigner();
+      this._providerType = walletProvider;
+      return;
     }
+    throw new Error(ERROR_MESSAGES.WALLET_TYPE_NOT_PROVIDED);
+  }
 
-    await this._walletConnectProvider.enable();
-    this._signer = new providers.Web3Provider(this._walletConnectProvider).getSigner();
+  private storeSession() {
+    if (this._runningInBrowser && window && this._didSigner) {
```

Comment:
Should this check for `window` should be included in the `runningInBrowser` check? Or is it only relevant in this case and in `clearSession`?

Reply:
Yeah, good point, we don't need to check if `window` is there cause the `runningInBrowser` is checking for env, any every browser has a `window`.